src/components/App.tsx

```diff
@@ -20,14 +21,48 @@ function App() {
   // State for the line height on the left
   const [lineHeightLeft, setLineHeightLeft] = useState<number>(1.5);
 
+  // Opentype feature option names from the gsub table of the font file on the left
+  const [fontFeatureOptionsLeft, setFontFeatureOptionsLeft] = useState<unknown[]>([]);
+
+  // Opentype feature option names from the gsub table of the font file on the right
+  const [fontFeatureOptionsRight, setFontFeatureOptionsRight] = useState<unknown[]>([]);
+
   // Handler for when a font is selected on the left side
   const handleFontSelectedLeft = (selectedFont: File | null) => {
     setSelectedFontLeft(selectedFont);
+    if (selectedFont != null) {
+      const buffer = selectedFont.arrayBuffer();
+      buffer.then((data) => {
+        const otfFont = opentype.parse(data);
+        const featureNames = [
+          ...Array.from(new Set(otfFont.tables.gsub.features.map((f: any) => f.tag))),
+        ];
+        //TODO: Remove debug logging of feature names
+        for (const name of featureNames) {
+          console.log(name);
+        }
+        setFontFeatureOptionsLeft(featureNames);
+      });
+    }
   };
 
   // Handler for when a font is selected on the right side
   const handleFontSelectedRight = (selectedFont: File | null) => {
     setSelectedFontRight(selectedFont);
+    if (selectedFont != null) {
+      const buffer = selectedFont.arrayBuffer();
+      buffer.then((data) => {
+        const otfFont = opentype.parse(data);
+        const featureNames = [
+          ...Array.from(new Set(otfFont.tables.gsub.features.map((f: any) => f.tag))),
+        ];
+        //TODO: Remove debug logging of feature names
```

Comment:
Buffer implementation looking good. 
Will these TODO be done in a separate PR? Can they be removed here?

Reply:
Done!